### ISSUE TITLE
Search term combinations returning too many results crash the app

### ISSUE THREAD

**Issue by j-marvin-GDS** [alphagov/govuk-knowledge-graph-gcp#477]
![image](https://github.com/alphagov/govuk-knowledge-graph-gcp/assets/94964460/4b274005-1cbd-4a3b-9273-97d2ffc3c9c2)

The app can crash when search combinations which return too many results are used. 

For example, searching for 'uk driving license' in the staging env (no quotes) will cause the environment to exhaust allocated memory. 

The following entry is left in the cloud log: 

`FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory`

**Comment by somme** [comment]
This is only an issue with staging/prototype as I have introduced a new field `text` which is all the body copy on that particular page. Looks like it runs out of memory as the can be quite large. To fix the issue I have added a query that extracts a paragraph instead of the entire text. The query looks for the first instance of the first keyword and trims left and right with 50 chars padding. This has solved the issue.
